src/ORM/ResultSetFactory.php

```diff
@@ -37,12 +37,19 @@ class ResultSetFactory
      * @param array $results Results array.
      * @return \Cake\ORM\ResultSet<array|\Cake\Datasource\EntityInterface>
      */
-    public function createResultSet(SelectQuery $query, array $results): ResultSet
+    public function createResultSet(SelectQuery $query, iterable $results, bool $bufferedResults = true): ResultSet
```

Comment:
Is `bufferedResults` used?

Reply:
I'll need to check again if that argument is needed. It's been a while since I worked on this so have forgotten some details :)